assets/js/modules/analytics/datastore/profiles.js

```diff
@@ -98,12 +99,13 @@ const baseActions = {
 	 * @since 1.8.0
 	 *
 	 * @param {string} propertyID Google Analytics property ID.
+	 * @param {string} profileName The name for a new profile.
 	 * @return {Object} Object with `response` and `error`.
 	 */
-	*createProfile( propertyID ) {
+	*createProfile( propertyID, profileName = '' ) {
```

Comment:
After thinking about this a bit, let's make `profileName` required on the client side.

The reason is that if a user cleared the input for the profile name and hit submit, they would end up with a new view using the default (from the server) but would not see the warning about the duplicate view name if they already had one.

Apart from requiring the parameter, we should add `invariant` for it here, as well as adding logic for it into the `canSubmitChanges` selector, so that if `PROFILE_CREATE` is chosen and no `profileName` is set it returns `false`.

Reply:
Updated.